### PR TITLE
migrate from drone to gh actions

### PR DESCRIPTION
**Pull request by ibrokethecloud**
Migration from drone to GH actions.

### CODE REVIEW

**Comment by ibrokethecloud** [comment]
related to: https://github.com/harvester/harvester/issues/5327

**Review by Yu-Jack — COMMENTED** [review]
Same as others, we should generate amd64 and arm64 build, and move specified binary in Dockerfile.

https://github.com/rancher/support-bundle-kit/blob/master/scripts/build
https://github.com/rancher/support-bundle-kit/blob/master/package/Dockerfile

**Review by Yu-Jack — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/template-build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Run dapper
+        run: make ci
+
+      - name: Read some Secrets
+        uses: rancher-eio/read-vault-secrets@main
+        if: ${{ inputs.push == true }}
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
```
Comment (FrankYang0529) [inline .github/workflows/template-build.yml:38]: Not sure whether we need to change `rancher` to `harvester`, but we can merge this first and fix it if need.

https://github.com/harvester/terraform-provider-harvester/blob/4e07ed02cba6fe7fc37bc0d41ff433df81e69915/.github/workflows/build.yml#L58-L59

File: .github/workflows/template-build.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        uses: docker/build-push-action@v5
+        with:
+          provenance: false
+          context: .
+          platforms: linux/amd64,linux/arm64
+          file: package/Dockerfile
+          push: ${{ inputs.push }}
+          tags: ${{env.imageName}}:${{ inputs.release-tag-name }}
+          build-args: |
+            VERSION=${{ github.ref_name }}-${{ github.sha }}-head
```
Comment (FrankYang0529) [inline .github/workflows/template-build.yml:57]: Is this version need to follow `release-tag-name`, so we don't get `<branch>-<sha>-head` for a tag version?

**Review by FrankYang0529 — COMMENTED** [review]
Overall LGTM. We may need to update `VERSION` in `build-args`. Thanks.

File: .github/workflows/template-build.yml (hunk 2 above)
Comment (Yu-Jack) [reply .github/workflows/template-build.yml:57]: I realized VERSION is not used in Dockerfile as build args, maybe we can remove it?

Comment (ibrokethecloud) [reply .github/workflows/template-build.yml:57]: good spot.. removed it.

the branch sha based tags are not pushed to dockerhub so i assume that is immaterial.

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thank you.

File: .github/workflows/template-build.yml (hunk 1 above)
Comment (ibrokethecloud) [reply .github/workflows/template-build.yml:38]: `rancher` is the correct org

**Review by bk201 — APPROVED** [review]

**Pull request merged**